### PR TITLE
feat(cobra): Use Cobra CLI framework for flag parsing and help text

### PR DESCRIPTION
**Pull request by ttomsu**
I started looking at github.com/spf13/cobra after @brandonnelson3 suggested it. This is a draft PR that I'm looking for feedback on. I believe I saw Kubernetes using this in the `kubectl`source code.

This would eliminate multiple `main.go` binaries, and everything would be under a single `rotation` command. The possible values, as structured here, would be:

```
rotation
rotation schedule
rotation schedule generate
rotation schedule extend
rotation calendar
rotation calendar sync
```  

Sample help doc:

```
$ go run rotation.go schedule generate
Error: required flag(s) "stop" not set
Usage:
  rotation schedule generate [flags]

Flags:
  -g, --github strings          Fetch the user list from GitHub. Order of args must be 'organization,team,accessToken'. Must specify an access token with read:org permissions.
  -h, --help                    help for generate
  -r, --record                  Record the responses from external dependencies to a file. Used for external dependency testing.
  -d, --shiftDurationDays int   Optional. Duration in days for each shift. Defaults to 7, must be a positive integer. (default 7)
      --start string            Optional. Generate schedule starting on this date (inclusive). Defaults to tomorrow's date. Must be in the format 2006-01-02
      --stop string             Required. Generate schedule stopping on this date (inclusive). Must be in the format 2006-01-02
  -u, --users strings           Set of users for the rotation. Required if --github* options are not specified.

```

My thinking is that this may help when we go to Dockerize this into an Action and the command becomes the thing we can tweak.

WDYT?

### CODE REVIEW

File: cmd/calendar.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package cmd
```
Comment (maggieneterval) [inline cmd/calendar.go:1]: Was briefly reading about Cobra before reviewing this PR and was just curious if you used the [Cobra Generator](https://github.com/spf13/cobra#using-the-cobra-generator) to auto-generate these files or if you created the recommended directory structure by hand?

File: cmd/extend.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+var extendCmd = &cobra.Command{
+	Use:   "extend",
+	Short: "Extends a previously generated schedule",
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("Extending schedule")
```
Comment (maggieneterval) [inline cmd/extend.go]: Guessing we would fill this out with something like the `RunE:  execute` of `generate.go`?

**Review by maggieneterval — COMMENTED** [review]
Seems reasonable to me! 🐍 

I know you mentioned this is a draft PR so I am selecting "comment" instead of "approve" so we can look at this again when it's ready. Will definitely be helpful to see the removed code the Cobra additions replace side-by-side, too.

**Comment by plumpy** [comment]
Seems reasonable to me... FYI `spin` uses `cobra` too...

**Comment by ezimanyi** [comment]
Seems reasonable to me too!

File: cmd/calendar.go (hunk 1 above)
Comment (ttomsu) [reply cmd/calendar.go:1]: I didn't - this was just done by hand.

File: cmd/extend.go (hunk 2 above)
Comment (ttomsu) [reply cmd/extend.go]: Yep, that's the intent - `Run` and `RunE` are the same, with `RunE` simply returning an `error`

**Comment by ttomsu** [comment]
Ok, I think I mostly like how this turned out.

A single `main` binary helps a lot of with shared functions. You could definitely argue that everything in one package spread over several files makes this harder in some respects, but I think that's true of Go in general.

**Comment by ttomsu** [comment]
I've moved the files out of their original directories since it's all one binary now. It's all in the second commit.

**Review by maggieneterval — APPROVED** [review]
🐍

**Review by ezimanyi — APPROVED** [review]

**Pull request merged**